Update Person class to use PayRate instead of EmploymentType

h4ow3i opened the pull request:
In this commit, the Person class has been refactored to replace the existing EmploymentType field with a new PayRate field. 

This change aligns with the evolving requirements of the application, where pay rates are now being tracked instead of employment types. 

The commit also includes relevant updates to method signatures, data parsing logic, and associated tests to accommodate the transition from employment types to pay rates.

h4ow3i committed:
Update Person class to use PayRate instead of EmploymentType

In this commit, the Person class has been refactored to replace the existing
EmploymentType field with a new PayRate field. This change aligns with the
evolving requirements of the application, where pay rates are now being
tracked instead of employment types. The commit also includes relevant updates
to method signatures, data parsing logic, and associated tests to accommodate
the transition from employment types to pay rates.

jxunze commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -29,7 +29,7 @@ public class AddCommand extends Command {
             + PREFIX_LASTNAME + "LAST NAME "
             + PREFIX_PHONE + "PHONE "
             + PREFIX_SEX + "SEX "
-            + PREFIX_EMPLOYMENTTYPE + "EMPLOYMENT TYPE "
```
can add space behind

h4ow3i replied:
fixed

jxunze commented on src/main/java/seedu/address/logic/commands/AddCommand.java:
```diff
@@ -38,7 +38,7 @@ public class AddCommand extends Command {
             + PREFIX_LASTNAME + "Doe "
             + PREFIX_PHONE + "98765432 "
             + PREFIX_SEX + "m "
-            + PREFIX_EMPLOYMENTTYPE + "ft "
```
ditto

h4ow3i replied:
fixed

jxunze commented on src/main/java/seedu/address/ui/PersonCard.java:
```diff
@@ -59,7 +59,7 @@ public PersonCard(Person person, int displayedIndex) {
         name.setText(person.getName().value);
         phone.setText(person.getPhone().value);
         sex.setText(getFullSexString(person.getSex().value));
-        employmentType.setText(getFullEmploymentTypeString(person.getEmploymentType().value));
```
hmm, can change? to .value? follow the standard from getName and getPhone

h4ow3i replied:
As mentioned, value of PayRate is a double, hence need to use toString method

jxunze commented on src/main/java/seedu/address/ui/PersonCard.java:
```diff
@@ -59,7 +59,7 @@ public PersonCard(Person person, int displayedIndex) {
+        payRate.setText(person.getPayRate().toString());
         address.setText(person.getAddress().value);
         bankDetails.setText(person.getBankDetails().value);
         hoursWorked.setText(person.getWorkHours().toString());
```
ditto

jxunze commented on src/test/data/JsonSerializableAddressBookTest/invalidPersonAddressBook.json:
```diff
@@ -4,7 +4,7 @@
       "firstName": "Alice",
       "lastName": "Pauline",
       "sex": "invalidsex",
-      "employmentType": "ft",
```
can standardise the parsing? I think parse as string better.

h4ow3i replied:
changed to double

jxunze commented on src/test/data/JsonSerializableAddressBookTest/typicalPersonsAddressBook.json:
```diff
@@ -5,7 +5,7 @@
       "firstName": "Alice",
       "lastName": "Pauline",
       "sex": "f",
-      "employmentType": "ft",
```
ditto

h4ow3i replied:
fixed

jxunze commented on src/test/java/seedu/address/testutil/TypicalPersons.java:
```diff
@@ -35,7 +35,7 @@ public class TypicalPersons {
         .withFirstName("Alice")
         .withLastName("Pauline")
         .withSex("f")
-        .withEmploymentType("ft")
```
I think store as string bah? or do u think int better

h4ow3i replied:
i think standardise payrate as double

h4ow3i committed:
Apply requested changes from PR review

This commit implements the changes requested during the PR review.

Specifically, updates have been made to several classes to address
issues raised in the review process. These changes ensure better
code quality and alignment with project requirements.
